internal/header/accept_charset.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package header
+
+import (
+	"sort"
+	"strings"
+)
+
+// AcceptCharset represents the Accept-Charset header.
+//
+// The "Accept-Charset" header field can be sent by a user agent to
+// indicate what charsets are acceptable in textual response content.
+// This field allows user agents capable of understanding more
+// comprehensive or special-purpose charsets to signal that capability
+// to an origin server that is capable of representing information in
+// those charsets.
+type AcceptCharset struct {
```

Comment:
can consider having `type AcceptCharset []Charset` instead, as the struct simply stores this single property.

Reply:
done!